Census/Listeners/CensusListener.cs

```diff
@@ -57,67 +59,96 @@ private async System.Threading.Tasks.Task StartConsumerLoop(CancellationToken ca
+                catch(ConsumeException ex)
                 {
-                    var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
-                    _logger.LogError(errorMessage);
+                    _nonTransientExceptionHandler.HandleException(ex);
+                    _logger.LogError($"Error consuming message: {ex.Error.Reason}");
                     continue;
                 }
 
-                deserializedMessage = message;
-                messageMetaData = ExtractFacilityIdAndCorrelationIdFromMessage(rawmessage.Message);
+                IBaseMessage deserializedMessage = null;
+                (string facilityId, string correlationId) messageMetaData = (string.Empty, string.Empty);
 
-                if (string.IsNullOrWhiteSpace(messageMetaData.facilityId))
+                if (rawmessage != null)
                 {
-                    var errorMessage = $"No Facility ID provided. Unable to process message: {message}";
-                    _logger.LogWarning(errorMessage);
-                    continue;
-                }
+                    IBaseMessage message = null;
+                    try
+                    {
+                        message = DeserializeMessage(rawmessage.Topic, rawmessage.Message.Value);
+                    }
+                    catch (Exception ex)
+                    {
+                        _nonTransientExceptionHandler.HandleException(rawmessage,ex);
+                        var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
+                        _logger.LogError(errorMessage);
+                        continue;
+                    }
 
-                try
-                {
-                    responseMessages = message switch
+                    deserializedMessage = message;
+                    messageMetaData = ExtractFacilityIdAndCorrelationIdFromMessage(rawmessage.Message);
+
+
+                    try
                     {
-                        PatientIDsAcquired => await _mediator.Send(new ConsumePatientIdsAcquiredEventCommand
+                        if (string.IsNullOrWhiteSpace(messageMetaData.facilityId))
                         {
-                            FacilityId = messageMetaData.facilityId,
-                            Message = (PatientIDsAcquired)message
-                        }, cancellationToken),
-                        _ => null
-                    };
+                            throw new MissingFacilityIdException("No Facility ID provided. Unable to process message.");
+                        }
+                    }
+                    catch (MissingFacilityIdException ex)
+                    {
+                        _nonTransientExceptionHandler.HandleException(rawmessage, ex);
+                        var errorMessage = $"No Facility ID provided. Unable to process message: {message}";
+                        _logger.LogWarning(errorMessage);
```

Comment:
Hey Ed, the event needs to be committed here after the nontransient handles the exception

Reply:
added